PlaceAutocomplete.md

```diff
@@ -0,0 +1,218 @@
+user searching for a place:
+
+1. A session token UUID `A` is generated for the session.
+2. The user begins typing a place name.
+3. On e.g. the third keystroke `"Har"` a request is sent to `/core/geography/place/autocomplete` with 
+`input`: `Har` and `session_token`: `A`.
+4. A list of predictions and their `place_id`s are generated for `"Har"` and returned in the response.
+5. The user does not select any of the predictions and instead inputs another keystroke `"Harb"`.
+6. `input`: `"Harb"` and `session_token`: `A` is sent in a new request.
+7. A new list of predictions and their `place_id`s are generated and returned
+8. The user selects one of the predictions.
+9. The `place_id` for the selected prediction is sent to the Place Details service along with 
+`session_token`: `A`.
+10. Place Details returns geographic details (locality name, lat/lng, etc.) for the selected place.
+11. A new session token is generated for any subsequent searches done by that user or any other
+users (i.e. DO NOT reuse session tokens).
+
+Using session tokens in this way greatly minimizes billing costs incurred. Please see [Google's session token
+documentation](https://developers.google.com/places/web-service/session-tokens) for more information.
+ 
+## Request Structure
+
+Request structure for this service mirrors [Google's request structure](https://developers.google.com/places/web-service/autocomplete#place_autocomplete_requests).
+
+#### Request
+
+|Field|Type|Required|Example|Note|
+|----|----|----|----|----|
+|`input`|`string`|True|`"Harbi"`||
+|`session_token`|`string`|True|-|Must be UUID.|
+|`offset`|`number`|False|`2`|Must be greater than 0 and less than input lenght|
+|`location`|`object`|False|-|See below for structure|
+|`origin`|`object`|False|-|See below for structure|
+|`radius`|`number`|False|`5000`|Meters|
+|`strict_bounds`|`boolean`|False|`true`||
+|`language`|`string`|False|`"et"`|See [supported languages](https://developers.google.com/maps/faq#languagesupport)|
+|`components`|`array[string]`|False|`["US", "CA", "CN"]`|No more than 5 countries. Must be ISO 3166-1 Alpha-2|
+|`types`|`string`|False|`"(region)"`|See [place types](https://developers.google.com/places/web-service/autocomplete#place_types)|
+
+#### `location` and `origin`
+|Field|Type|Required|Example|Note|
+|----|----|----|----|----|
+|`latitude`|`number`|True|45.7576||
+|`longitude`|`number`|True|126.6409||
+
+#### Examples
+
+Only required fields:
+````json
+{
+    "input": "Harbi",
+    "session_token": "<UUID_TOKEN>"
+}
+````
+
+All fields:
+````json
+{
+    "input": "Harbi",
+    "session_token": "<UUID_TOKEN>",
+    "offset": 4.0,
+    "location": {
+        "latitude": 45.7576,
+        "longitude": 126.6409
+    },
+    "origin": {
+        "latitude": 45.818107,
+        "longitude": 126.599279
+    },
+    "radius": 50000,
+    "strict_bounds": true,
+    "language": "zh-cn",
+    "components": ["CN", "US", "CA", "ET", "FR", "UK"],
```

Comment:
The request in service limit this one to 5 or less then 5 entries

Reply:
Fixed.